module/zfs/zfs_vnops.c

```diff
@@ -3010,29 +3009,27 @@ zfs_setattr(struct inode *ip, vattr_t *vap, int flags, cred_t *cr)
 
 	if (mask & ATTR_MTIME) {
 		ZFS_TIME_ENCODE(&vap->va_mtime, mtime);
+		ZTOI(zp)->i_mtime = timespec_trunc(vap->va_mtime,
+				ZTOI(zp)->i_sb->s_time_gran);
```

Comment:
Strance, my cstyle doesn't report anything, that's why I pushed it. Are you using a modified cstyle or something?